### ISSUE TITLE
extended kafka-connect:0.16.2-kafka-2.4.0 not loading plugins?

### ISSUE THREAD

**Issue by jslusher** [strimzi/strimzi-kafka-operator#2517]
I have been extending the strimzi-connect image in my connect cluster to incorporate a few plugins (debezium, elasicsearch-connect, jdbc-mysql) and it was working fine using my previous version. I recently upgraded my connect images from using `strimzi-connect:0.14.0-kafka-2.3.0` to `0.16.2-kafka-2.4.0` and the connect pods are giving errors about not having the plugins anymore. I can see the plugins in `/opt/kafka/plugins`, so I'm not sure where else to look.

```
[kafka@platform-connect-55459b8ffd-ksvpf kafka]$ ls -l /opt/kafka/plugins/
total 4
drwxr-xr-x 6  501 games  110 Nov 12 23:57 confluentinc-kafka-connect-elasticsearch-5.3.1
drwxr-xr-x 1  501 games   45 Feb  7 23:27 confluentinc-kafka-connect-jdbc-5.3.2
drwxr-xr-x 2 root root  4096 Feb  7 23:26 debezium-connector-mysql
```

```
2020-02-08 00:05:18,847 ERROR Uncaught exception in REST call to /connectors/my-threads/status (org.apache.kafka.connect.runtime.rest.errors.ConnectExceptionMapper) [qtp1517758613-22]
org.apache.kafka.connect.errors.ConnectException: Failed to find any class that implements Connector and which name matches io.debezium.connector.mysql.MySqlConnector, available connectors are: PluginDesc{klass=class org.apache.kafka.connect.file.FileStreamSinkConnector, name='org.apache.kafka.connect.file.FileStreamSinkConnector', version='2.4.0', encodedVersion=2.4.0, type=sink, typeName='sink', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.file.FileStreamSourceConnector, name='org.apache.kafka.connect.file.FileStreamSourceConnector', version='2.4.0', encodedVersion=2.4.0, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.mirror.MirrorCheckpointConnector, name='org.apache.kafka.connect.mirror.MirrorCheckpointConnector', version='1', encodedVersion=1, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.mirror.MirrorHeartbeatConnector, name='org.apache.kafka.connect.mirror.MirrorHeartbeatConnector', version='1', encodedVersion=1, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.mirror.MirrorSourceConnector, name='org.apache.kafka.connect.mirror.MirrorSourceConnector', version='1', encodedVersion=1, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.MockConnector, name='org.apache.kafka.connect.tools.MockConnector', version='2.4.0', encodedVersion=2.4.0, type=connector, typeName='connector', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.MockSinkConnector, name='org.apache.kafka.connect.tools.MockSinkConnector', version='2.4.0', encodedVersion=2.4.0, type=sink, typeName='sink', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.MockSourceConnector, name='org.apache.kafka.connect.tools.MockSourceConnector', version='2.4.0', encodedVersion=2.4.0, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.SchemaSourceConnector, name='org.apache.kafka.connect.tools.SchemaSourceConnector', version='2.4.0', encodedVersion=2.4.0, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.VerifiableSinkConnector, name='org.apache.kafka.connect.tools.VerifiableSinkConnector', version='2.4.0', encodedVersion=2.4.0, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.VerifiableSourceConnector, name='org.apache.kafka.connect.tools.VerifiableSourceConnector', version='2.4.0', encodedVersion=2.4.0, type=source, typeName='source', location='classpath'}
```

I see similar exceptions for the other plugins as well, so it's not just one plugin.

**Comment by scholzj** [comment]
For whatever reason the ownership of some of the plugins looks weird: `501 games`. I wonder if it has something to do with it.

**Comment by jslusher** [comment]
The ownership of a couple of them do look strange now that you mention it, but the debezium plugin should still work (root:root) and it doesn't seem to. Also the permissions on all the folders are world-executable, and I would think that should be good enough. So you are saying the they are where they should be? Is there an environment variable to check for `CLASSPATH` or similar? I tried, but there doesn't seem to be a value for `CLASSPATH`.

**Comment by jslusher** [comment]
After further investigating I must have gotten my pods mixed up and I didn't notice that the operator was changing my temporarily set `imagePullPolicy` back to `IfNotPresent`. And so there was one pod that had the plugins in the proper place, but the other two did not. After I stopped the operator and set `imagePullPolicy` to `Always`, I was able to roll the pods and they picked up the fixed image that unfortunately had the same tag as the image that did not have the files in the right place. Thanks for the reply!
